universal-application-tool-0.0.1/app/services/applicant/question/ApplicantQuestion.java

```diff
@@ -139,28 +153,32 @@ public boolean hasErrors() {
   }
 
   public Optional<Long> getUpdatedInProgramMetadata() {
-    Path contextualizedMetadataPath = getContextualizedPath().join(Scalar.PROGRAM_UPDATED_IN);
+    return getMetadata(Scalar.PROGRAM_UPDATED_IN);
+  }
 
-    // Metadata for enumerators is stored for each JSON array element, but we rely on metadata for
-    // the first one.
-    if (getQuestionDefinition().isEnumerator()) {
-      contextualizedMetadataPath =
-          getContextualizedPath().atIndex(0).join(Scalar.PROGRAM_UPDATED_IN);
-    }
+  public Optional<Long> getLastUpdatedTimeMetadata() {
+    return getMetadata(Scalar.UPDATED_AT);
+  }
 
+  private Optional<Long> getMetadata(Scalar metadataScalar) {
+    Path contextualizedMetadataPath = getMetadataPath(metadataScalar);
     return applicantData.readLong(contextualizedMetadataPath);
   }
 
-  public Optional<Long> getLastUpdatedTimeMetadata() {
-    Path contextualizedMetadataPath = getContextualizedPath().join(Scalar.UPDATED_AT);
+  private Path getMetadataPath(Scalar metadataScalar) {
+    Path contextualizedMetadataPath = getContextualizedPath().join(metadataScalar);
```

Comment:
super nit (non-blocking): could this just be an if/else? We don't use the `contextualizedMetadataPath` in the if statement

Reply:
duh. fixed!